clients/src/main/java/org/apache/kafka/common/utils/Bytes.java

```diff
@@ -40,6 +41,26 @@ public static Bytes wrap(byte[] bytes) {
         return new Bytes(bytes);
     }
 
+    /**
+     * Create a Bytes using the byte buffer. If the provided byteBuffer contains the whole content, we can directly
+     * use the backed array. If the byteBuffer has only partial of the content (ex: a sliced byteBuffer), we'll do array copy
+     *
+     * @param byteBuffer    The byteBuffer becomes the backing storage for the object.
+     */
+    public static Bytes wrap(ByteBuffer byteBuffer) {
```

Comment:
Create a `Bytes.wrap` for byteBuffer instance, and see if we can re-use the backed array directly, or we need to do array copy.